components/builder-api/src/server/handlers.rs

```diff
@@ -187,8 +194,17 @@ pub fn validate_docker_credentials(req: &mut Request) -> IronResult<Response> {
         return Ok(Response::with(status::BadRequest));
     }
 
-    // There's probably a better way to do this. Suggestions?
-    let client = match ApiClient::new("https://hub.docker.com/v2", PRODUCT, VERSION, None) {
+    let url = match body["url"].as_str() {
+        Some(url) => url,
+        None => {
+            match registry_type.as_ref() {
+                "docker" => "https://hub.docker.com/v2",
```

Comment:
Should this be `docker-hub` (or similar), since you can run your own Docker registry?